internal/testutil/helpers.go

```diff
@@ -20,8 +20,8 @@ var DevZero io.Reader = devZero{}
 type devZero struct{}
 
 func (d devZero) Read(p []byte) (n int, err error) {
-	for i := 0; i < len(p); i++ {
-		p[i] = '\x00'
+	for i := range p {
+		p[i] = 0
```

Comment:
but `\x00` is `NULL`, not `0`?

Reply:
`'\x00'` is a rune which results in a byte having all bits set to zero. `0` is an integer constant having all bits set to zero. These two have the same effect when assigned to a `byte` (see https://play.golang.org/p/IUboS5HCYkI). Now sure what do you mean by `NULL`, but if there would be `NULL` in Go as it is in C, it will also be equal to `0` and `\x00` (although, if I remember it correctly, this is not guaranteed by the C language specification, but is true in practice for every implementation known). Now, `nil` might be equivalent, too, but due to strict[er than C] type checking it can not be assigned to a `byte`.

The code before and after this patch is equivalent in what it's doing, but somehow Go fails to optimize the binary code for the old version, and optimizes the new one pretty good, as described in patch commit message.

Reply:
Oh, thanks! I for some reason thought `NULL` would be different than `0`; shows what I know :blush: